docroot/themes/humsci/su_humsci_theme/patterns/molecules/generic-three-column/generic-three-column.html.twig

```diff
@@ -0,0 +1,41 @@
+{% set attributes = attributes.addClass('generic-three-column', variant|clean_class) %}
+
+{% set leftAttributes = create_attribute({class:'left-region'}) %}
+{% set middleAttributes = create_attribute({class:'middle-region'}) %}
+{% set rightAttributes = create_attribute({class:'right-region'}) %}
+
+{% if 'static' in variant %}
+  {% set static = true %}
+{% endif %}
+
+{% if '25_50_25' in variant %}
+  {% set middleAttributes = middleAttributes.addClass('larger') %}
+{% elseif '50_25_25' in variant %}
+  {% set leftAttributes = leftAttributes.addClass('larger') %}
+{% elseif '25_25_50' in variant %}
+  {% set rightAttributes = rightAttributes.addClass('larger') %}
+{% endif %}
+
+{% spaceless %}
+  <div{{ attributes }}>
+
+    {%- if left|has_markup or static -%}
+      <div{{ leftAttributes }}>
+        {{ left }}
+      </div>
+    {%- endif -%}
+
+    {%- if middle|has_markup or static -%}
```

Comment:
Although I like this filter is it really necessary? Why is the https://twig.symfony.com/doc/2.x/filters/striptags.html filter not enough?

Reply:
we'd have to specify every time all the tags that would present valid content. it would have to be something like 
`middle|strip_tags("<img> <iframe> <object> <picture> <figure>...")` for every time we want to check

Reply:
So in this case, you would have to do it three times? And you would gain the flexibility and granularity of specifying those tags for each instance? I get the DRY thinking of this but I think it adds something to maintain when you already have a tool to use.

Reply:
in the future if we discover a tag that is appropriate, we'd have to add it to every line where this would be used. hyptothetically if we left out `<iframe>` and realize its valid markup, we'd have to manually add it to every line, not just this one patter (when we use it more)
Also i dont think its much to maintain. its a 1 line static method vs several lines in twigs. :man_shrugging:

Reply:
Yeah, it is a trade for sure. We need more opinions. 

@boznik @jbickar may we have your thoughts on this.

Reply:
the two options are 
1. have a special twig filter that can be referenced with `foo|has_markup`
  * Small code to maintain
2. Use a built in twig filters like `foo|render|striptags('<img> <a> <iframe> <object> <picture> <figure>')|trim` for each condition check.

Reply:
i'm inclined to go with pookmish on this one. maybe we should put this in a seperate twig helper module where we could all reuse special filters across different sites?

Reply:
watcha think @sherakama @jbickar ?

Reply:
I like that idea. I am worried that we will be creating templates that cannot be shared across our projects and will be dependant on modules. Let's start a project to contain these twig plugins and require it in our themes.

Reply:
so should this get merged  and moved to a separate module later?

Reply:
Yeah, I don't want to hold this up any longer. Go forth and refactor later.

Reply:
I've added a topic to our next dev chat to go through this.